Base/Models/Public/Include/Models/Models.hpp

```diff
@@ -0,0 +1,110 @@
+#pragma once
+#include "PostgreModel.hpp"
+#include <FileLogger.hpp>
+#include <map>
+#include <utility>
+
+namespace Models
+{
+using Base::Logger::FileLogger;
+
+
+enum class UserInfo : uint16_t
+{
+    ID = 0,
+    EMAIL,
+    LOGIN,
+    PASSHASH
+};
+
+template<template<typename ...TArgs> class RepositoryType, class TEnum = UserInfo, typename... Args>
```

Comment:
A high level of indirectness. Try to simplify

Reply:
Going to be reworked soon